golem_sci/factory.py

```diff
@@ -64,11 +75,10 @@ def new_sci(
     _ensure_genesis(web3, chain)
     provider = ContractDataProvider(chain)
     geth_client = Client(web3)
-    nonce = geth_client.get_transaction_count(address)
     return SCIImplementation(
         geth_client,
         address,
-        JsonTransactionsStorage(datadir, nonce),
+        storage,
```

Comment:
@Krigpl 
Concent's implementation of storage (`JsonTransactionStorage` as well) gets `nonce` as parameter, maybe better way would be to pass as parameter nonce to storage?